### PR TITLE
Remove any references to media.

### PR DESCRIPTION
**Pull request by kixelated**
Going to be moved to another draft that uses tracks.
Will rename the file to `draft-moq-transport` after this gets merged.

### CODE REVIEW

File: draft-lcurley-warp.md [hunk 1]
```diff
@@ -87,89 +78,62 @@ either directly or via WebTransport {{WebTransport}}.
...
-: Packet loss and queuing caused by degraded or overloaded networks.
-
-Consumer:
+: A QUIC endpoint: either a client or a server.
 
-: A QUIC endpoint receiving media over the network. This could be the media player or middleware.
+Session:
 
-Container:
+: An QUIC connection identified via a URL after a handshake.
```
Comment (vasilvv) [inline draft-lcurley-warp.md:87]: I'm not sure this is necessarily true -- you can have multiple WT sessions on a QUIC connection, for instance.  Was there anything wrong specifically with the old transport session definition?

File: draft-lcurley-warp.md [hunk 2]
```diff
@@ -87,89 +78,62 @@ either directly or via WebTransport {{WebTransport}}.
...
+Server:
 
-: An video image or group of audio samples to be rendered at a specific point in time.
+: The endpoint that accepts incoming QUIC connections and Warp sessions.
 
-I-frame:
+Publisher:
 
-: A frame that does not depend on the contents of other frames; effectively an image.
+: An endpoint that publishes tracks and transmits objects over the network.
```
Comment (vasilvv) [inline draft-lcurley-warp.md:99]: "Publisher is an endpoint that publishes" -- that is a bit of a circular definition.  Maybe, since we always require subscribing, we could use something along the lines of "publisher is an endpoint that fulfills the subscription requests and sends objects?"

File: draft-lcurley-warp.md [hunk 3]
```diff
@@ -87,89 +78,62 @@ either directly or via WebTransport {{WebTransport}}.
...
-: A QUIC endpoint sending media over the network. This could be the media encoder or middleware.
+: An entity that uses Warp to send or receive live data.
 
-Server:
-
-: The party accepting an incoming Warp session.
+Track:
 
-Slice:
+: A live stream, fragmented into objects that are generated over time.
```
Comment (vasilvv) [inline draft-lcurley-warp.md:123]: I feel like "a live stream" is ambiguous.  A sequence of objects?  A stream of objects?  A feed of objects?

File: draft-lcurley-warp.md [hunk 4]
```diff
@@ -87,89 +78,62 @@ either directly or via WebTransport {{WebTransport}}.
...
+Subscriber:
 
-: A I-frame followed by a sequential series of dependent frames.
+: An endpoint that requests tracks and receives objects over the network.
 
-Group of samples:
+Producer:
 
-: A sequential series of audio samples starting at a given timestamp.
+: The initial publisher that encodes tracks and objects.
```
Comment (suhasHere) [inline draft-lcurley-warp.md:107]: I would either refer to the terminology defined in https://datatracker.ietf.org/doc/html/draft-nandakumar-moq-arch-00#section-2 or copy some sort of these.

Since there is security role played by each entity and it will be good to define along those lines.

File: draft-lcurley-warp.md [hunk 5]
```diff
@@ -87,89 +78,62 @@ either directly or via WebTransport {{WebTransport}}.
...
+: A live stream, fragmented into objects that are generated over time.
 
-: A section of a video frame. There may be multiple slices per frame.
+Group:
 
-Track:
+: A collection of objects, used to denote join points for new subscribers.
 
-: An encoded bitstream, representing a single media component (ex. audio, video, subtitles) that makes up the larger broadcast.
+Object:
```
Comment (suhasHere) [inline draft-lcurley-warp.md:130]: group/object -> prefer add a reference to the model section

File: draft-lcurley-warp.md [hunk 6]
```diff
@@ -207,9 +171,9 @@ In general, objects within a group SHOULD NOT depend on objects in other groups.
 
 ## Track {#model-track}
 
-A Track is the central concept within the MoQ Transport protocol for delivering media and is made up of sequence of objects ({{model-object}}) organized in the form of groups ({{model-group}}).
-
-A track is a transform of a uncompressed media or metadata using a specific encoding process, a set of parameters for that encoding, and possibly an encryption process. The MoQ Transport protocol is designed to transport tracks.
+A *track* is a sequence of objects ({{model-object}}) organized into groups ({{model-group}}).
+A subscriber can request individual tracks at group boundaries, including any new objects while the track is active.
+The application is responsible for determining how tracks are named, encoded, or otherwise used.
```
Comment (suhasHere) [inline draft-lcurley-warp.md:176]: i don't think the last statement is needed.

File: draft-lcurley-warp.md [hunk 7]
```diff
@@ -219,25 +183,8 @@ Tracks are identified by a globally unique identifier, called "Full Track Name"
...
-Track Namespace = livestream.example.com
-Track Name = uaCafDkl123/audio
-Full Track Name = livestream.example.com/uaCafDkl123/audio
-
-Example: 3
-Track Namespace = security-camera.example.com/camera1
-Track Name = hd-video
-Full Track Name = security-camera.example.com/camera1/hd-video
-
-~~~
```
Comment (suhasHere) [inline draft-lcurley-warp.md:240]: Let's please add the examples back.

File: draft-lcurley-warp.md [hunk 8]
```diff
@@ -250,121 +197,66 @@ The client issues a CONNECT request with a URL which the server uses for identif
...
-We can generally classify existing media protocols into two categories based on the underlying network protocol:
-
-* TCP-based media protocols (ex. RTMP, HLS, DASH) are popular due to their simplicity.
-Media is served/consumed in decode order while any networking is handled by the TCP layer.
-However, these protocols primarily see usage at higher latency targets due to their relatively slow detection and response to queuing.
-
-* UDP-based media protocols (ex. RTP, WebRTC, SRT) can side-step the issues with TCP and provide lower latency with better queue management.
-However the media protocol is now responsible for fragmentation, congestion control, retransmissions, receiver feedback, reassembly, and more.
-This added complexity significantly raises the implementation difficulty and hurts interoperability.
+The speed at which a live protocol can detect and respond to queuing determines the latency.
```
Comment (suhasHere) [inline draft-lcurley-warp.md:215]: s/live protocol/moq-transport protocol

Live protocol seems bit ambigouos or partly defined.

File: draft-lcurley-warp.md [hunk 9]
```diff
@@ -250,121 +197,66 @@ The client issues a CONNECT request with a URL which the server uses for identif
...
+TCP-based protocols are simple, but are slow to detect congestion and suffer from head-of-line blocking.
+UDP-based protocols can avoid queuing, but the application is now responsible for fragmentation, congestion control, retransmissions, receiver feedback, reassembly, and more.
 
-A goal of this draft is to get the best of both worlds: a simple protocol that can still rapidly detect and respond to congestion.
-This is possible with the emergence of QUIC, designed to fix the shortcomings of TCP.
+A goal of this draft is to get the best of both worlds: a simple protocol that can still rapidly detect and respond to congestion using QUIC streams.
 
 ## Universal
-The media protocol ecosystem is fragmented; each protocol has it's own niche.
+The live media protocol ecosystem is fragmented; each protocol has it's own niche.
```
Comment (suhasHere) [inline draft-lcurley-warp.md:222]: s/live media protocol/moq-transport protocol

File: draft-lcurley-warp.md [hunk 10]
```diff
@@ -250,121 +197,66 @@ The client issues a CONNECT request with a URL which the server uses for identif
...
+The goal is not to reinvent how content is encoded, just delivered.
 
 ## Relays
 The prevailing belief is that UDP-based protocols are more expensive and don't "scale".
 While it's true that UDP is more difficult to optimize than TCP, QUIC itself is proof that it is possible to reach performance parity.
-In fact even some TCP-based protocols (ex. RTMP) don't "scale" either and are exclusively used for contribution as a result.
 
-The ability to scale a media protocol actually depends on relay support: proxies, caches, CDNs, SFUs, etc.
-The success of HTTP-based media protocols is due to the ability to leverage traditional HTTP CDNs.
+The ability to scale a protocol actually depends on relay support: proxies, caches, CDNs, SFUs, etc.
```
Comment (suhasHere) [inline draft-lcurley-warp.md:237]: I feel by remove the term media, we are loosing clarity here.
to scale a "to scale a live media protocol " is still fine. Media doesn't have to me audio or video. Or may be reword it as "to scale a live content delivery transport"

File: draft-lcurley-warp.md [hunk 11]
```diff
@@ -702,7 +582,7 @@ The SETUP parameters are described in the {{setup-parameters}} section.
 
 
 ## OBJECT {#message-object}
-A OBJECT message contains a single media object associated with a specified track, as well as associated metadata required to deliver, cache, and forward it.
+A OBJECT message contains a byte slice associated with a specified track, as well as associated metadata required to deliver, cache, and forward it.
```
Comment (suhasHere) [inline draft-lcurley-warp.md:585]: A OBJECT message contains a byte slice or A OBJECT message contains chunk of data ?

**Review by suhasHere — CHANGES_REQUESTED** [review]
Thanks for the PR .. Few nits on terminology and suggestions on some.

File: draft-lcurley-warp.md [hunk 12]
```diff
@@ -1,5 +1,5 @@
 ---
-title: "Warp - Live Media Transport over QUIC"
+title: "Warp - Live Transport over QUIC"
 abbrev: WARP
```
Comment (suhasHere) [inline draft-lcurley-warp.md:3]: Pleas rename WARP to MoQTransport .

s/Warp - Live Transport over QUIC/MoQTransport - Live  Content Transport Protocol over QUIC

**Review by suhasHere — CHANGES_REQUESTED** [review]

File: draft-lcurley-warp.md (hunk 8 above)
Comment (afrind) [reply draft-lcurley-warp.md:215]: Since this is about motivation, I don't think using moq-transport-protocol here is necessary.  Maybe just

`The speed at which the system can detect and respond...?`

File: draft-lcurley-warp.md (hunk 9 above)
Comment (afrind) [reply draft-lcurley-warp.md:222]: I think `live media protocol ecosystem` is correct here, or maybe just `media protocol ecosystem`

File: draft-lcurley-warp.md [hunk 13]
```diff
@@ -250,121 +197,66 @@ The client issues a CONNECT request with a URL which the server uses for identif
...
-
-A goal of this draft is to cover a large spectrum of use-cases. Specifically:
-
-* Consolidated contribution and distribution.
-The primary difference between the two is the ability to fanout.
-How does a CDN know how to forward media to N consumers and how does it reduce the encoded bitrate during congestion?
-A single protocol can cover both use-cases provided relays are informed on how to forward and drop media.
+For example, a service might simultaneously ingest via WebRTC, SRT, RTMP, etc.
+The same service might then simultaneously distribute via WebRTC, LL-HLS, HLS/DASH, etc.
+And then distribute other live content over other protocols: for example updates, chat, metadata, etc.
```
Comment (afrind) [inline draft-lcurley-warp.md:227]: This is not a sentence.

File: draft-lcurley-warp.md [hunk 14]
```diff
@@ -250,121 +197,66 @@ The client issues a CONNECT request with a URL which the server uses for identif
...
 
-* A configurable latency versus quality trade-off.
-The producer (broadcaster) chooses how to encode and transmit media based on the desired user experience.
-Each consumer (viewer) chooses how long to wait for media based on their desired user experience and network.
-We want an experience that can vary from real-time and lossy for one viewer, to delayed and loss-less for another viewer, without separate encodings or protocols.
-
-A related goal is to not reinvent how media is encoded.
-The same codec bitstream and container should be usable between different protocols.
+This draft attempts to build a unified base live transport for media and similar use-cases.
```
Comment (afrind) [inline draft-lcurley-warp.md:229]: s/base live transport/base transport protocol/ ?

File: draft-lcurley-warp.md [hunk 15]
```diff
@@ -559,10 +441,10 @@ The application MAY use any error message and SHOULD use a relevant code, as def
 |------|--------------------|
 
 * Session Terminated
-No error occured, however the endpoint no longer desires to send or receive media.
+No error occurred, however the endpoint wishes to terminate.
```
Comment (afrind) [inline draft-lcurley-warp.md:444]: terminate the session.

File: draft-lcurley-warp.md (hunk 11 above)
Comment (afrind) [reply draft-lcurley-warp.md:585]: Byte slice came out of nowhere.  Maybe 

`An OBJECT message contains a range of contiguous bytes from a track, ...`

Though this now begs the questions about whether Object metadata should also carry the offset into the track (maybe?, optionally?), and whether data from multiple OBJECTs can overlap (hopefully not).

File: draft-lcurley-warp.md [hunk 16]
```diff
@@ -953,15 +789,15 @@ The ROLE parameter (key 0x00) allows the client to specify what roles it expects
 
 0x01:
 
-: Only the client is expected to send media on the connection. This is commonly referred to as *the ingestion case*.
+: Only the client is expected to send object on the connection. This is commonly referred to as *the ingestion case*.
```
Comment (afrind) [inline draft-lcurley-warp.md:792]: objects

**Review by afrind — COMMENTED** [review]
Individual/editorial review

File: draft-lcurley-warp.md (hunk 12 above)
Comment (suhasHere) [reply draft-lcurley-warp.md:3]: The comment indicates file name rename .. May be my confusion?

File: draft-lcurley-warp.md (hunk 4 above)
Comment (suhasHere) [reply draft-lcurley-warp.md:107]: that would be fine too and also bringing something as informational reference shouldn't be bad either

File: draft-lcurley-warp.md (hunk 2 above)
Comment (suhasHere) [reply draft-lcurley-warp.md:99]: This is what is currently defined in the arch spec
"Entities with Publisher role publish MoQ Objects belonging to a given track using the MoQ Transport Protocol."

Comment (suhasHere) [reply draft-lcurley-warp.md:99]: May be we can do it in its own PR. Created an issue to discuss the same here: https://github.com/kixelated/warp-draft/issues/157

File: draft-lcurley-warp.md (hunk 4 above)
Comment (fluffy) [reply draft-lcurley-warp.md:107]: +1 one on getting all the definitions this draft needs into this draft.

File: draft-lcurley-warp.md (hunk 3 above)
Comment (fluffy) [reply draft-lcurley-warp.md:123]: To help the reader understand, I think it would be useful to give and an example along lines of "a track would often be used to represent a single stream of video, or audio". Something that ties it back to media and helps the reader understand what this might be.

File: draft-lcurley-warp.md [hunk 17]
```diff
@@ -87,89 +78,62 @@ either directly or via WebTransport {{WebTransport}}.
...
+Track:
 
-Slice:
+: A live stream, fragmented into objects that are generated over time.
 
-: A section of a video frame. There may be multiple slices per frame.
+Group:
 
-Track:
+: A collection of objects, used to denote join points for new subscribers.
```
Comment (fluffy) [inline draft-lcurley-warp.md:127]: Might be good to add in that a group is part of a single track.

File: draft-lcurley-warp.md [hunk 18]
```diff
@@ -250,121 +197,66 @@ The client issues a CONNECT request with a URL which the server uses for identif
...
-As an example, consider a scenario where `example.org` hosts a simple live stream that anyone can subscribe to.
-That live stream would be a single track bundle, accessible via the WebTransport URL: `https://example.org/livestream`.
-In a simple scenario, the track bundle would contain only two media tracks, one with audio and one with video.
-In a more complicated scenario, the track bundle could multiple tracks with different formats, encodings, bitrates, and quality levels, possibly for the same content.
-The receiver learns about each available track within the bundle via the catalog, and can choose to subscribe to a subset.
-
 
 # Motivation
 
 ## Latency
```
Comment (fluffy) [inline draft-lcurley-warp.md:203]: FWIW ... in this section, I think it would be fine to talk about media instead of content but I am fine with it either way

File: draft-lcurley-warp.md [hunk 19]
```diff
@@ -528,11 +410,11 @@ As covered in the motivation section ({{motivation}}), the ability to prioritize
 It's equally important to detect congestion via congestion control, which is handled in the QUIC layer {{QUIC-RECOVERY}}.
 
 Bufferbloat is caused by routers queueing packets for an indefinite amount of time rather than drop them.
```
Comment (fluffy) [inline draft-lcurley-warp.md:412]: I might rephrase this as something like .... "Bufferbloat occurs when routers queue packets for too long instead of dropping the packet which can add significant latency."

**Review by fluffy — APPROVED** [review]
I added some small comments which can be incorporated or ignored as you see fit. I think this does a good job of the the main goal of moving th media part to  the other draft.

File: draft-lcurley-warp.md (hunk 18 above)
Comment (kixelated) [reply draft-lcurley-warp.md:203]: I'd like to move this motivation section to an overview or requirements document, and yeah make it media-specific.

File: draft-lcurley-warp.md (hunk 7 above)
Comment (kixelated) [reply draft-lcurley-warp.md:240]: Sorry, I got carried away.

Comment (kixelated) [reply draft-lcurley-warp.md:240]: FYI the examples could be shorter.

```
/Users/kixelated/work/warp-draft/draft-lcurley-warp.xml(182): Warning: Artwork too wide, reducing indentation from 3 to 0
/Users/kixelated/work/warp-draft/draft-lcurley-warp.xml(170): Warning: Section too wide, reducing indentation from 0 to 0
/Users/kixelated/work/warp-draft/draft-lcurley-warp.xml(165): Warning: Section too wide, reducing indentation from 0 to 0
/Users/kixelated/work/warp-draft/draft-lcurley-warp.xml(140): Warning: Section too wide, reducing indentation from 0 to 0
/Users/kixelated/work/warp-draft/draft-lcurley-warp.xml(66): Warning: Middle too wide, reducing indentation from 0 to 0
s/Users/kixelated/work/warp-draft/draft-lcurley-warp.xml(182): Warning: Too long line found (L286), 8 characters longer than 72 characters:
Track Namespace = videoconferencing.example.com/meetings/m123/participants/alice
/Users/kixelated/work/warp-draft/draft-lcurley-warp.xml(182): Warning: Too long line found (L288), 14 characters longer than 72 characters:
Full Track Name = videoconferencing.example.com/meetings/m123/participants/alice/audio
```

File: draft-lcurley-warp.md (hunk 2 above)
Comment (kixelated) [reply draft-lcurley-warp.md]: I agree, but I removed the changes to the definitions anyway.

File: draft-lcurley-warp.md (hunk 4 above)
Comment (kixelated) [reply draft-lcurley-warp.md]: I removed any changes to the definitions because it was out of place.

off-topic: I wouldn't define the producer as the entity with encryption keys, especially because that's optional. The producer's main role is to package live content into tracks/groups/objects. There should definitely be a note saying the producer SHOULD be the only entity with encryption keys, but that access does not define it.

File: draft-lcurley-warp.md (hunk 12 above)
Comment (kixelated) [reply draft-lcurley-warp.md:3]: I made a separate PR to do the rename (#161), and it looks like Github can show the diff correctly.

Comment (suhasHere) [inline draft-lcurley-warp.md:3]: As agreed on the authors call, lets keep the word "media" here

File: draft-lcurley-warp.md [hunk 20]
```diff
@@ -45,40 +45,32 @@ normative:
...
-    date: 2020-03
   NewReno: RFC6582
   BBR: I-D.cardwell-iccrg-bbr-congestion-control-02
 
 --- abstract
 
-This document defines the core behavior for Warp, a live media transport protocol over QUIC.
-Media is split into objects based on the underlying media encoding and transmitted independently over QUIC streams.
-QUIC streams are prioritized based on the send order, allowing less important objects to be starved or dropped during congestion.
+This document defines the core behavior for Warp, a live transport protocol over QUIC.
```
Comment (suhasHere) [inline draft-lcurley-warp.md:55]: As agreed on the authors call, lets keep the word "media" here

File: draft-lcurley-warp.md [hunk 21]
```diff
@@ -45,40 +45,32 @@ normative:
...
+The application fragments a live stream into objects, including a header that describes the basic relationship between objects.
+Objects are starved/dropped during congestion based on priorities in order to minimize latency.
 
 --- middle
 
 
 ## Introduction
-Warp is a live media transport protocol that utilizes the QUIC network protocol {{QUIC}},
```
Comment (suhasHere) [inline draft-lcurley-warp.md:71]: As agreed on the authors call, lets keep the word "media" here

File: draft-lcurley-warp.md [hunk 22]
```diff
@@ -45,40 +45,32 @@ normative:
...
+Warp is a live transport protocol that utilizes the QUIC network protocol {{QUIC}},
 either directly or via WebTransport {{WebTransport}}.
+It was originally developed for live media, but has been generalized for similar use-cases.
 
 * {{motivation}} covers the background and rationale behind Warp.
-* {{objects}} covers how media is fragmented into objects.
+* {{objects}} covers how data is fragmented into objects.
```
Comment (suhasHere) [inline draft-lcurley-warp.md:68]: s/data/application data/

File: draft-lcurley-warp.md [hunk 23]
```diff
@@ -45,40 +45,32 @@ normative:
...
 * {{transport-protocols}} covers aspects of setting up a MoQ transport session.
-* {{stream-mapping}} covers how QUIC is used to transfer media.
+* {{stream-mapping}} covers how QUIC is used to transfer data.
```
Comment (suhasHere) [inline draft-lcurley-warp.md:70]: s/data/application data

File: draft-lcurley-warp.md [hunk 24]
```diff
@@ -219,7 +155,9 @@ Tracks are identified by a globally unique identifier, called "Full Track Name"
 Full Track Name = Track Namespace  "/"  Track Name
 ~~~~~~~~~~~~~~~
 
-This document does not define the exact mechanism of naming Track Namespaces. Applications building on top of MoQ MUST ensure that the mechanism used guarantees global uniqueness; for instance, an application could use domain names as part of track namespaces. Track Namespace is followed by the application context specific Track Name, encoded as an opaque string. 
+This document does not define the exact mechanism of naming Track Namespaces. Applications building on top of MoQ MUST ensure that the mechanism used guarantees global uniqueness; for instance, an application could use domain names as part of track namespaces. Track Namespace is followed by the application context specific Track Name, encoded as an opaque string.
+
+This document does not define the exact mechanism of naming Track Namespaces. Applications building on top of MoQ MUST ensure that the mechanism used guarantees global uniqueness; for instance, an application could use domain names as part of track namespaces. Track Namespace is followed by the application context specific Track Name, encoded as an opaque string.
```
Comment (suhasHere) [inline draft-lcurley-warp.md:161]: duplicate text

**Review by suhasHere — APPROVED** [review]
